src/runtime/internal/dom.ts

```diff
@@ -630,6 +630,17 @@ export function add_resize_listener(node: HTMLElement, fn: () => void) {
 	};
 }
 
+export function add_resize_observer(node: HTMLElement, fn: (arg: any) => void) {
+	if ('ResizeObserver' in window) {
+		// @ts-ignore https://github.com/Microsoft/TypeScript/issues/28502
+		const observer = new ResizeObserver(entries => fn(entries.values().next().value));
+		observer.observe(node);
+		return () => observer.disconnect();
+	} else {
+		return noop;
+	}
+}
```

Comment:
this probably shouldn't fail silently — if `ResizeObserver` isn't present, we should throw a useful error in dev, otherwise it should just fail with a `ReferenceError`